header-validator/src/validate-json.ts

```diff
@@ -565,6 +568,29 @@ function aggregationKeys(ctx: Context, j: Json): Maybe<Map<string, bigint>> {
   )
 }
 
+function clamp<N extends bigint | number>(
+  ctx: Context,
+  n: N,
+  min: N,
+  max: N
+): N {
+  if (n < min) {
+    ctx.warning(`will be clamped to min of ${min}`)
+    return min
+  }
+  if (n > max) {
+    ctx.warning(`will be clamped to max of ${max}`)
+    return max
+  }
+  return n
+}
+
+function expiry(ctx: Context, j: Json): Maybe<number | bigint> {
+  return legacyDuration(ctx, j).map((n) =>
+    clamp(ctx, n, limits.sourceExpiryRange[0], limits.sourceExpiryRange[1])
```

Comment:
Just a note that, for event source, the expiry is rounded first before clamping, but this is fine for now.

Reply:
Yes, I intend to add the rounding in a followup.